assets/prototype/domain/eventEditor/tickets/ticketPriceCalculator/hooks/useTicketPriceCalculatorFormDecorator.js

```diff
@@ -36,89 +18,11 @@ const getBasePriceType = (price, priceTypes) => find(propEq('id', price.priceTyp
-	const updateBasePrice = useCallback(
-		(formData) => {
-			const result = calculator(formData, { type: 'CALCULATE_BASE_PRICE' });
-			const newBasePrice = getValue('prices[0].amount', result);
-			return { ['prices[0].amount']: parseFloat(newBasePrice || 0) };
-		},
-		[calculator]
-	);
-	const updateTicketTotal = useCallback(
-		(formData) => {
-			const result = calculator(formData, { type: 'CALCULATE_TICKET_TOTAL' });
-			const newTicketTotal = getValue('ticket.price', result);
-			return { ['ticket.price']: parseFloat(newTicketTotal || 0) };
-		},
-		[calculator]
-	);
-	return createDecorator(
-		{
-			field: /^prices\[\d+\]\.amount$/,
-			isEqual: amountsEqual,
-			updates: (value, name, formData) => {
-				const reverseCalc = getValue('ticket.reverseCalculate', formData);
-				if (reverseCalc) {
-					if (name === 'prices[0].amount') {
-						return {};
-					}
-					return updateBasePrice(formData);
-				}
-				return updateTicketTotal(formData);
-			},
-		},
-		{
-			field: /^prices\[\d+\]\.priceType$/,
-			isEqual: isEqual,
-			updates: (value, name, formData) => {
-				const pricePath = name.replace('.priceType', '');
-				const price = getValue(pricePath, formData);
-				const priceType = getBasePriceType(price, priceTypes);
-				const updatedPrice = {
-					...price,
-					isDiscount: priceType.isDiscount,
-					isPercent: priceType.isPercent,
-					order: priceType.order,
-					priceType: priceType.id,
-				};
-				const updatedPrices = map(
-					when(propEq('id', updatedPrice.id), () => updatedPrice),
-					formData.prices
-				);
-				const formDataWithUpdatedPrice = { ticket: formData.ticket, prices: updatedPrices };
-				const reverseCalc = getValue('ticket.reverseCalculate', formDataWithUpdatedPrice);
-				const formDataCalculations = reverseCalc
-					? updateBasePrice(formDataWithUpdatedPrice)
-					: updateTicketTotal(formDataWithUpdatedPrice);
-				return {
-					...formDataCalculations,
-					[pricePath]: updatedPrice,
-				};
-			},
-		},
-		{
-			field: 'ticket.reverseCalculate',
-			isEqual: boolsEqual,
-			updates: (value, name, formData) => {
-				const reverseCalc = getValue('ticket.reverseCalculate', formData);
-				return reverseCalc ? updateBasePrice(formData) : updateTicketTotal(formData);
-			},
-		},
-		{
-			field: 'ticket.price',
-			isEqual: amountsEqual,
-			updates: (value, name, formData) => {
-				const reverseCalc = getValue('ticket.reverseCalculate', formData);
-				// we don't want to update the base price if reverse calculate is false
-				if (reverseCalc) {
-					return updateBasePrice(formData);
-				}
-				return {};
-			},
-		}
-	);
+	const priceAmountDecorator = usePriceAmountDecorator();
+	const priceTypeDecorator = usePriceTypeDecorator();
+	const reverseCalculateDecorator = useReverseCalculateDecorator();
+	const ticketTotalDecorator = useTicketTotalDecorator();
+	return createDecorator(priceAmountDecorator, priceTypeDecorator, reverseCalculateDecorator, ticketTotalDecorator);
```

Comment:
Cool 👍

Reply:
Cool indeed, perfect size for a file :)
![image](https://user-images.githubusercontent.com/1477453/72079978-98859b00-3304-11ea-88d7-a154da742b21.png)